src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -1342,6 +1448,64 @@ public static async Task<Stream> ReadAsStreamUncompressedAsync(HttpResponseMessa
             return true;
         }
 
+        private bool ShouldBroadcast<TResult>(RavenCommand<TResult> command)
+        {
+            if (command is IBroadcast == false)
+                return false;
+
+            if (TopologyNodes == null || 
+                TopologyNodes.Count < 2)
+                return false;
+
+            return true;
+        }
+
+        private async Task<TResult> Broadcast<TResult>(RavenCommand<TResult> command, SessionInfo sessionInfo, CancellationToken token)
+        {
+            if (command is IBroadcast == false)
+                throw new InvalidOperationException("You can broadcast only commands that implement 'IBroadcast'.");
+
+            var tasks = new Dictionary<Task, (RavenCommand<TResult> Command, int Index)>();
+            command.FailedNodes = new Dictionary<ServerNode, Exception>(); // clear the current failures
+
+            for (var index = 0; index < _nodeSelector.Topology.Nodes.Count; index++)
+            {
+                var node = _nodeSelector.Topology.Nodes[index];
+
+                var returnContext = ContextPool.AllocateOperationContext(out JsonOperationContext ctx);
+
+                var clone = (RavenCommand<TResult>)((IBroadcast)command).PrepareToBroadcast(ctx, Conventions);
+                clone.SetTimeout(_raftBroadcastTimeout);
+
+                var task = ExecuteAsync(node, null, ctx, clone, shouldRetry: false, sessionInfo, token);
+#pragma warning disable 4014
+                task.ContinueWith(_ => returnContext.Dispose(), TaskContinuationOptions.ExecuteSynchronously);
+#pragma warning restore 4014
+
+                tasks.Add(task, (clone, index));
+            }
+
+            while (tasks.Count > 0)
+            {
+                var completed = await Task.WhenAny(tasks.Keys).ConfigureAwait(false);
+                if (completed.IsCanceled || completed.IsFaulted)
+                {
+                    var failed = tasks[completed];
+                    var node = _nodeSelector.Topology.Nodes[failed.Index];
+                    command.FailedNodes[node] = completed.Exception.ExtractSingleInnerException();
+                    _nodeSelector.OnFailedRequest(failed.Index);
+
+                    tasks.Remove(completed);
+                    continue;
+                }
+                _nodeSelector.RestoreNodeIndex(tasks[completed].Index);
+                return tasks[completed].Command.Result;
```

Comment:
What happens to the other tasks?
Maybe we can pass `LinkedToken` and cancel them.

Reply:
Or better, record it as a failed request.
If we have the same command, will it succeed on all of the nodes? (if they are online)

Reply:
We need to cancel the others, I think.
Important, we have to be sure that we dispose the ctx properly.